tests/integration/tasks/sync.ts

```diff
@@ -1,10 +1,42 @@
+import * as chalk from 'chalk';
 import { handler } from '../../../src/commands/tasks/sync';
 import { runtimeChoices } from '../../../src/constants';
+import { functionCode } from '../../__helpers__/functions';
+import { tasksRepositoryMock } from '../../__helpers__/tasksRepositoryMock';
+import { createTempDirectoryManager } from '../../__helpers__/tempDirectoryManager';
 
 describe('exh tasks sync', () => {
+  let tempDirectoryManager;
+  let repositoryMock;
+
+  beforeAll(async () => {
+    tempDirectoryManager = await createTempDirectoryManager();
+    repositoryMock = tasksRepositoryMock();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterAll(async () => {
+    await tempDirectoryManager.removeDirectory();
+  });
+
   const root = 'tests/__helpers__/task-configs/invalid-runtimes/';
   const runtimes = runtimeChoices.map(runtime => runtime).join(', ');
 
+  it('Creates a Function', async () => {
```

Comment:
(optional) Can you also create a test that updates a function?  That would trigger the bug in the find spy.